extra_geom/detectors.py

```diff
@@ -1014,6 +1014,90 @@ class DSSC_1MGeometry(DetectorGeometryBase):
         [0.5, 1, 1, 0.5, 0, 0]
     ])
 
+    @classmethod
+    def from_quad_positions(cls, quad_pos, *, unit=1e-3, asic_gap=None,
+                            panel_gap=4):
+        """Generate a DSSC-1M geometry from quadrant positions.
+
+        This produces an idealised geometry, assuming all modules are perfectly
+        flat, aligned and equally spaced within their quadrant.
+
+        The position given should refer to the bottom right (looking
+        along the beam) corner of the quadrant.
+
+        The origin of the coordinates is in the centre of the detector.
+        Coordinates increase upwards and to the left (looking along the beam).
+
+        Parameters
+        ----------
+        quad_pos: list of 2-tuples
+          (x, y) coordinates of the last corner (the one by module 4) of each
+          quadrant.
+        unit: float, optional
+          The conversion factor to put the coordinates into metres.
+          The default 1e-3 means the numbers are in millimetres.
+        asic_gap: float, optional
+          The gap between adjacent tiles/ASICs. The default is 2 mm.
+        panel_gap: float, optional
+          The gap between adjacent modules/panels. The default is 4 mm.
+        """
+        assert len(quad_pos) == 4
+        asic_gap_m = 2e-3 if (asic_gap is None) else asic_gap * unit
```

Comment:
IMO, that's not quite obvious: in the function definition we say `asic_gap=None` and `panel_gap=4` but we _implicitly_ define the default `asic_gap` is not `None` but two mm. While with the `panel_gap` value we define the default value explicitly (well, we do another trick which leads to that) as four mm. Maybe there is a reason behind this way of definition, but otherwise I'd think about the way of defining both `asic_gap` and `panel_gap` in a more, so to say, symmetric way.

Reply:
Good catch, thanks. They should both be None, so that if you pass `unit=`, the default gaps are still 2 mm & 4 mm, not 2 & 4 of whatever unit you specify.